README.md

```diff
@@ -1,12 +1,201 @@
 # Box
+
 [![codecov](https://codecov.io/gh/jan-xyz/box/branch/main/graph/badge.svg?token=261hqevLHY)](https://codecov.io/gh/jan-xyz/box)
+[![Go Reference](https://pkg.go.dev/badge/github.com/jan-xyz/box.svg)](https://pkg.go.dev/github.com/jan-xyz/box)
 
 Boxing everything in neat little boxes, stack them, combine them, hide them.
 
-Very much inspired by [go-kit](https://github.com/go-kit/kit)
+The goal of this project is to provide simple abstractions that avoid leaking
+models across boundaries and provide tools to create applications that are convenient
+to reason about and maintainable with a string compatibility with existing libraries.
+
+The main idea stems from my experience with [go-kit](https://github.com/go-kit/kit)
+and it inspired the majority of this project. However, over the years I noticed
+some things where go-kit was deviating from standard tools, and with the release
+generics in Go, I thought about providing a type-safe version of what go-kit taught
+me.
+
+## Philosophy
+
+The basic idea is that most applications have to deal with different model layers
+when they interface with the outside world:
+
+1. transport protocol - e.g. HTTP requests/responses, async message buses,
+  gRPC, lambda trigger etc.
+1. data-transfer-object (DTO) - Protocolbuffers, JSON, AVRO, database models
+1. internal model - the domain model used inside of the application
+
+For each of the models there is a distinct layer that provides conversion logic
+domain specific handling and encapsulation.
+
+Terms used in this applications to refer to the different layers are:
+
+| Layer | Model | Concerns |
+|-------|--------|---------|
+| Handler | transport protocol | Communication patterns, status codes, meta-data extraction, error conversion |
```

Comment:
I know we chose that internally but I would reconsider the usage of `Handler` here. In hindsight I find `transport` a way better fit for what the layer is actually doing as every layer is 'handling' something.

Reply:
Good point, does it make sense to refer to the entire thing as the handler when thinking about AWS, and also the standard library HTTP server also uses [handler](https://pkg.go.dev/net/http)?